extensions/arrow/src/main/java/io/deephaven/extensions/arrow/sources/AbstractArrowColumnSource.java

```diff
@@ -0,0 +1,74 @@
+package io.deephaven.extensions.arrow.sources;
+
+import io.deephaven.base.verify.Assert;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.table.SharedContext;
+import io.deephaven.engine.table.impl.AbstractColumnSource;
+import io.deephaven.extensions.arrow.ArrowWrapperTools;
+import io.deephaven.util.QueryConstants;
+import io.deephaven.util.datastructures.LongSizedDataStructure;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.LongConsumer;
+
+/**
+ * Base class for arrow column sources
+ * 
+ * @param <T>
+ */
+public abstract class AbstractArrowColumnSource<T> extends AbstractColumnSource<T> {
```

Comment:
This feels *so* similar to the regioned/page code. I wonder if we could re-use that?

Reply:
I'm not sure we can use the regioned page code. A RecordBatch might have size up to MAX_ARRAY_LEN (if the feather file is playing nice to java-users); 2^20 is probably not large enough for arbitrary files.

Reply:
Regions max out much larger than 2^20.

Reply:
I've filed #3633 - I think this is a good idea, but after digging into the Parquet implementation I feel that we should do this only if we have users (or customers) who really need a more efficient implementation. For this, I think we lose most of the original DHE implementation, increases scope, and might make it difficult to meet our porting goal on time.